### DISCUSSION TITLE
Multi-artifacts: Secondary artifact not adding examples

### DISCUSSION THREAD

**Discussion by juaninicolai** [microcks/microcks#1259] [Q&A] [answered]
I have uploaded via UI an OpenAPI yaml file to mock an API. This works ok, but since the response data is very large, the API contract does not contains examples. Just metadata, info, paths and schemas.

After this, I've uploaded a Postman collection for the same API which contains some saved samples for different scenarios.
From the server logs I can see the postman collection gets successfully added as secondary artifact (I am checking that box on the upload modal) and I can see it in the microcks API Details.

But the Postman samples are not reflected on the 'operations' section on the UI. I can only see 'commodities' response (coming from the OpenAPI file) but not the samples included via Postman collection.

![Screenshot 2024-08-06 at 14 57 39](https://github.com/user-attachments/assets/f63ac34a-99a3-4879-95a8-c8f31042f75f)

What is the expectation about this? Should I see more response tabs / options available after adding the secondary artifact?

**Comment by yada** [comment]
Hi Juani,

Can you see the collection on the APIs services properties? You should have
your openAPI and you collection in contracts.

If not, then check:
https://microcks.io/documentation/explanations/multi-artifacts/
«

The merging process is based on a compound key: the API *name* + *version*.

If not explicitly identified as primary or secondary, the default is to
consider an imported artifact as the primary one. Microcks will simply
ignore a secondaryartifact if it doesn’t match any existing API name +
version. »

Regards,

Yacine
Le mar. 6 août 2024 à 14:10, Juani Nicolai ***@***.***> a
écrit :

**Reply by juaninicolai** [reply]
Hi Yacine, I can see both (postman collections and openAPI contract) on the properties. I have uploaded one of them as secondary following that article exactly :) (you can find those details in the attached picture).

But even though I can see both files in my mocked API, only one of them is reflecting the response samples.

Is there anything else besides API name + version I should consider to get this right?

Regards, 
Juan

**Reply by lbroudoux** [reply]
Hi there,

You may also check that your examples in the Postman Collection refer to the correct operation. The matching is done using the verb and URL mentioned.

Typically, you should have something like `GET http://{baseUrl}/commodities` in your example so that Microcks can assign it to a known operation. If no matching found, the example is ignored.

**Reply by lbroudoux** [reply]
Any news on this topic? If no progress done, do not hesitate sharing a small reproducer

**Reply by juaninicolai** [reply]
Hi @lbroudoux ,sorry for this late response. I have been busy working on other things.

Do you mean that the postman samples names should be verb + URL to match the API Specs examples?

**Reply by lbroudoux** [reply]
Yep. You could have a different host or prefix but verb and url suffix should match

**Reply by juaninicolai** [reply]
Oh okey, that's not what I have at the moment. I have API specs with verbs and paths and postman samples with some specific names indicating the flow it represents. 

I will test this tomorrow and come back to you. 
Thanks for the help.

**Reply by juaninicolai** [reply]
Hello @lbroudoux I was not able to get this done and sadly I can't share a reproducer for my API because it holds company data.

Would you be able to provide a Postman collection for the hello.yaml API you are using in the microcks documentation?
Just for testing / exploring purposes I have added some error responses to the file. 

Had to change the file format to txt to attached here.
[hello.txt](https://github.com/user-attachments/files/16622545/hello.txt)

Hopefully you can help me, thanks in advance!
Juan

**Reply by lbroudoux** [reply]
Yes. I'll try to do that on tomorrow (today's bank holiday ;-))

**Reply by juaninicolai** [reply]
Hello @lbroudoux please ping me back with that reproducer whenever you have time :)
Thank you for your help!

**Reply by lbroudoux** [reply, marked as answer]
You can have a look at this OpenAPI file: https://github.com/microcks/api-lifecycle/blob/master/contract-testing-demo/apipastries-openapi.yaml
and the corresponding Postman Collection that add 1 example on the `GET /pastry/{name}` operation: https://github.com/microcks/api-lifecycle/blob/master/contract-testing-demo/apipastries-postman-collection.json.

The matching is done thanks to [line 102](https://github.com/microcks/api-lifecycle/blob/master/contract-testing-demo/apipastries-postman-collection.json#L102) where `/pastry/:name` (+ verb on line 99) is matched with the [line 77](https://github.com/microcks/api-lifecycle/blob/master/contract-testing-demo/apipastries-openapi.yaml#L77) in OpenAPI + verb on line 79.

Hope it helps,

**Reply by juaninicolai** [reply]
Hello @lbroudoux with your example and this [article](https://microcks.io/documentation/references/artifacts/postman-conventions/) I was able to fix this. Thank you!
